src/overlays/actors/ovl_En_Ge2/z_en_ge2.c

```diff
@@ -24,65 +62,598 @@ const ActorInit En_Ge2_InitVars = {
+    if (func_8002F334(&this->actor, globalCtx)) {
+
+        switch (this->actor.params & 0xFF) {
+            case GE2_TYPE_PATROLLING:
+                EnGe2_ChangeAction(this, GE2_ACTION_ABOUTTURN);
+                break;
+            case GE2_TYPE_STATIONARY:
+                EnGe2_ChangeAction(this, GE2_ACTION_STAND);
+                break;
+            case GE2_TYPE_GERUDO_CARD_GIVER:
+                this->actionFunc = EnGe2_WaitLookAtPlayer;
+                break;
+        }
+        this->actor.update = EnGe2_UpdateFriendly;
+        this->actor.flags &= ~0x10000;
+    }
+    EnGe2_TurnToFacePlayer(this, globalCtx);
+}
+
+// Same as EnGe3_WaitLookAtPlayer
+void EnGe2_WaitLookAtPlayer(EnGe2* this, GlobalContext* globalCtx) {
+    EnGe2_LookAtPlayer(this, globalCtx);
+}
+
+// Same as EnGe3_WaitTillCardGiven
+void EnGe2_WaitTillCardGiven(EnGe2* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        this->actionFunc = EnGe2_SetActionAfterTalk;
+    } else {
+        func_8002F434(&this->actor, globalCtx, GI_GERUDO_CARD, 10000.0f, 50.0f);
+    }
+}
+
+// Same as EnGe3_GiveCard
+void EnGe2_GiveCard(EnGe2* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+        this->actor.flags &= ~0x10000;
+        this->actionFunc = EnGe2_WaitTillCardGiven;
+        func_8002F434(&this->actor, globalCtx, GI_GERUDO_CARD, 10000.0f, 50.0f);
+    }
+}
+
+// Similar, but not identical, to EnGe3_ForceTalk
+void EnGe2_ForceTalk(EnGe2* this, GlobalContext* globalCtx) {
+
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->actionFunc = EnGe2_GiveCard;
+    } else {
+        this->actor.textId = 0x6004;
+        this->actor.flags |= 0x10000;
+        func_8002F1C4(&this->actor, globalCtx, 300.0f, 300.0f, 0);
+    }
+    EnGe2_LookAtPlayer(this, globalCtx);
+}
+
+void EnGe2_SetupCapturePlayer(EnGe2* this, GlobalContext* globalCtx) {
+    this->stateFlags |= GE2_STATE_CAPTURING;
+    this->actor.speedXZ = 0.0f;
+    EnGe2_ChangeAction(this, GE2_ACTION_CAPTURETURN);
+    func_8002DF54(globalCtx, &this->actor, 95);
+    func_80078884(NA_SE_SY_FOUND);
+    func_8010B680(globalCtx, 0x6000, &this->actor);
+}
+
+// Same as EnGe3_MaintainCollider
+void EnGe2_MaintainColliderAndSetAnimState(EnGe2* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_8002E4B4(globalCtx, &this->actor, 40.0f, 25.0f, 40.0f, 5);
+
+    if ((!(this->stateFlags & GE2_STATE_ANIMCOMPLETE)) && SkelAnime_Update(&this->skelAnime)) {
+        this->stateFlags |= GE2_STATE_ANIMCOMPLETE;
+    }
+}
+
+// Same as EnGe3_MoveAndBlink
+void EnGe3_MoveAndBlink(EnGe2* this, GlobalContext* globalCtx) {
```

Comment:
```suggestion
// Same as EnGe3_MoveAndBlink
void EnGe2_MoveAndBlink(EnGe2* this, GlobalContext* globalCtx) {
```
Might be the same, but probably shouldn't be named the same.